packages/firestore/src/core/query.ts

```diff
@@ -110,7 +110,7 @@ export class Query {
   addFilter(filter: Filter): Query {
     assert(
       this.getInequalityFilterField() == null ||
-        !(filter instanceof RelationFilter) ||
+        !(filter instanceof FieldFilter) ||
         !filter.isInequality() ||
```

Comment:
In your interest of avoiding instanceof and over-constraining, we could promote isInequality() to `Filter` if we wanted. 🤷‍♂